modules/imgimport/src/csvreading.cpp

```diff
@@ -165,21 +210,44 @@ vector<Metadata> readcsv(const char* filename){
             }
             if(heads[j].compare("photonum")==0){
                 finput>>temnum;
-                if(temnum>=i){
+                if(temnum>photonum){
                     f=1;
                 }
                 j++;
                 continue;
             }
             cerr<<"header error"<<endl;
-            return errorResult;
+            return;
         }
+        mark++;
         if(f==1){
-            result.push_back(input);
-            i++;
+            log.push_back(input);
+            photonum++;
             f=0;
         }
     }
     finput.close();
-    return result;
+}
+
+Metadata metadata_input::get_metadata(double timestamp){
+    read_metadata();
+    for(int i=0;i<log.size();i++){
+        if(log.at(i).time==timestamp){
```

Comment:
Not only is comparing doubles using == not recommended, you will always want to return something unless there are no entries available. This function should be returning the _closest entry_ (ideally with some sort of extrapolation, as well as noting error caused by the potential difference between the timestamp wanted and the timestamp found, but that we can probably leave for now due to time constraints).